File: Class work/10-09-24/task2.c
#include<stdio.h>

prime(int n)
{
    int c=0,i;
    for(i=1;i<=n;i++)
    {
        if(n%i==0)
        c++;

    }
    if (c==2)
    {
        printf("this is a prime number");
    }
    else
    {
        printf("this is not a prime number");
    }
  
}


main()
{
    int n;

    printf("Enter the number: ");
    scanf("%d",&n);
    prime(n);
}

File: Assigment/python fundamental/task2.py
def is_prime(number):
    # Not prime if 1 or less
    if number <= 1:
        return False
    # Check factors from 2 to sqrt(number)
    for i in range(2, int(number ** 0.5) + 1):
        # Not prime if factor found
        if number % i == 0:
            return False
    # Prime if no factors found
    return True

# Example usage
num = int(input("Enter a number: "))
if is_prime(num):
    print(f"{num} is a prime number.")
else:
    print(f"{num} is not a prime number.")